src/components/blocks/blocks.js

```diff
@@ -0,0 +1,63 @@
+import styles from "@components/list/list.scss";
+import {commonHelpers} from "@/utils";
+import Icon from "@/icon";
+import React from "react";
+
+/**
+ *
+ * @param props and custom
+ * props contains:
+     * data: array,
+     * handleDetails: func,
+     * handleEdit: func,
+     * access: boolean (true by default),
+     * fieldsToShow: Array<str>
+ * custom: custom field
+ * @returns jsx in block view
+ */
+
+export const Blocks = ({data, handleDetails, handleEdit, access = true, fieldsToShow}, custom) =>
+    data.map(({
+                  id,
+                  index,
+                  firstName,
+                  lastName,
+                  email,
+                  name,
+                  themeName,
+                  lessonShortDate,
+                  lessonTime,
+                  studentIds,
+                  startDate,
+                  finishDate
+              }) =>
+        {
+            return <tr key={id}
+                       onClick={handleDetails? () => handleDetails(id): null}
```

Comment:
it could be better to remove into separate fn with useCallback , especially if it's ternary operator